### PR TITLE
feat: Adding a note with UBI recommendation / Devfile.io

### PR DESCRIPTION
**Pull request by ibuziuk**
Screenshot 2024-04-25 at 15 08 17

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

**Review by deerskindoll — APPROVED** [review]
lgtm

**Review by svor — APPROVED** [review]
lgtm

File: modules/end-user-guide/pages/devfile-introduction.adoc [hunk 1]
```diff
@@ -5,21 +5,24 @@ link:https://devfile.io/[Devfiles] are `yaml` text files used for development en
 
 .{prod}-specific devfile features
 
+[NOTE]
+====    
+{prod} is expected to work with most of the popular images defined in the `components` section of devfile. For production purposes, it is recommended to use one of the link:https://catalog.redhat.com/software/containers/search?gs&q=ubi[Universal Base Images] as a base image for defining the Cloud Development Environment.
```
Comment (RomanNikitenko) [inline modules/end-user-guide/pages/devfile-introduction.adoc:10]: some ubi images can not be used: 
- `ubi8-9` https://github.com/eclipse-che/che/issues/22870
- `ubi7`-based images were not tested

Comment (RomanNikitenko) [reply modules/end-user-guide/pages/devfile-introduction.adoc:10]: I don't know if this ^ should be mentioned, just letting you know about this limitation...

Comment (ibuziuk) [reply modules/end-user-guide/pages/devfile-introduction.adoc:10]: I think we should mention it, lemme update and add WARNING

**Comment by ibuziuk** [comment]
@RomanNikitenko PR has been updated please review

**Review by RomanNikitenko — APPROVED** [review]

**Comment by ibuziuk** [comment]
@deerskindoll could you please merge?

**Pull request merged**